src/pages/ProfilePage.js

```diff
@@ -0,0 +1,150 @@
+                            />
+                        </View>
+                        <Text style={[styles.settingsDisplayName, styles.mt1, styles.mb6]} numberOfLines={1}>
+                            {profileDetails.displayName
+                                ? profileDetails.displayName
+                                : null}
+                        </Text>
+                        {profileDetails.login && (
+                        <View style={[styles.mb6, styles.profilePageSectionContainer]}>
+                            <Text style={[styles.profilePageLabel, styles.mb2]} numberOfLines={1}>
+                                {Str.isSMSLogin(profileDetails.login) ? 'Phone Number' : 'Email'}
+                            </Text>
+                            <Text style={[styles.profilePageLabel]} numberOfLines={1}>
+                                {Str.isSMSLogin(profileDetails.login)
+                                    ? Str.removeSMSDomain(profileDetails.login)
+                                    : profileDetails.login}
+                            </Text>
+                        </View>
+                        )}
+                        {profileDetails.pronouns && (
+                        <View style={[styles.mb6, styles.profilePageSectionContainer]}>
+                            <Text style={[styles.profilePageLabel, styles.mb2]} numberOfLines={1}>
+                                Preferred Pronouns
+                            </Text>
+                            <Text style={[styles.profilePageLabel]} numberOfLines={1}>
+                                {profileDetails.pronouns}
+                            </Text>
+                        </View>
+                        )}
+                        {profileDetails.timezone && (
+                        <View style={[styles.mb6, styles.profilePageSectionContainer]}>
+                            <Text style={[styles.profilePageLabel, styles.mb2]} numberOfLines={1}>
+                                Local Time
+                            </Text>
+                            <Text style={[styles.profilePageLabel]} numberOfLines={1}>
+                                {moment().tz(profileDetails.timezone).format('LT')}
+                            </Text>
+                        </View>
+                        )}
+                    </View>
+                </View>
+
+                <Text style={[styles.profilePageSectionVersion]} numberOfLines={1}>
+                    v
+                    {version}
+                </Text>
+            </View>
+        </>
+    );
+};
+
+ProfilePage.propTypes = propTypes;
+ProfilePage.displayName = 'ProfilePage';
+
+
+export default compose(
+    withOnyx({
+        personalDetails: {
+            key: ONYXKEYS.PERSONAL_DETAILS,
+        },
+    }),
+    withOnyx({
+        currentlyViewedReportID: {
+            key: ONYXKEYS.CURRENTLY_VIEWED_REPORTID,
+        },
+    }),
+    withOnyx({
```

Comment:
`withOnyx()` can subscribe to multiple keys at once so we probably do not need 3 of these.
For the `currentlyViewedReportID` if we pass it as a match param instead then we can use it to fetch the report.

However, again, I am confused about why we are basing this on the report participants. It doesn't seem necessary to look at the report at all. If we are routing to `/profile/:login` then we need only look at the `:login` match param and then locate it in the `personalDetails`. 

IMO this solution is more complex since instead of getting the details by doing `this.props.personalDetails[this.props.match.params.login]` we are subscribing to three separate Onyx keys.

Please let me know if I'm missing something...

Reply:
interesting, I tried adding all the keys into one and it was complaining so I kept them separated but maybe I did something wrong. but as you mentioned, with the original implementation there would only need to be one key